File: Music.Client.Android.Short/app/src/main/java/com/zitherharpmusic/zhmshort/ui/item/ItemListFragment.java
package com.zitherharpmusic.zhmshort.ui.item;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.zitherharpmusic.zhmshort.R;
import com.zitherharpmusic.zhmshort.model.Music;

import java.util.List;

public class ItemListFragment extends Fragment {
    private static List<? extends Music> musics;

    public static ItemListFragment newInstance(List<? extends Music> musics) {
        ItemListFragment.musics = musics;
        return new ItemListFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_video_grid, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        RecyclerView recyclerView = view.findViewById(R.id.recycler_view);
        recyclerView.setAdapter(new ItemListAdapter(getActivity(), musics));
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
    }
}

File: Music.Client.Android/Kotlin/Store/app/src/main/java/com/zitherharp/store/ui/item/ItemListFragment.kt
package com.zitherharp.store.ui.item

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.zitherharp.store.R
import com.zitherharp.store.databinding.FragmentItemListBinding
import com.zitherharp.store.model.Item

class ItemListFragment(private val items: List<Item>) : Fragment() {
    private lateinit var binding: FragmentItemListBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?): View {
        return FragmentItemListBinding.inflate(inflater, container, false).apply {
            binding = this
        }.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        with(binding) {
            itemList.adapter = ItemListAdapter(view.context, items)
            if (items.isNotEmpty()) {
                status.text = ""
            } else {
                status.text = getString(R.string.empty)
            }
        }
    }
}